### ISSUE TITLE
Is there a way to make the formatter ignore reader macros?

### ISSUE THREAD

**Issue by smashedtoatoms** [nobody-famous/alive#39]
I use the `#+(or)` reader macro quite a bit to comment out expressions and the auto-formatter is formatting them like regular comments.  Any chance the formatted could ignore them?

**Comment by nobody-famous** [comment]
Yes, that should be doable. Do you have an example of what it's doing versus what it should do?

**Comment by smashedtoatoms** [comment]
Totally.  __Here are some examples of how it should look:__

```lisp
(defun test (a &optional b)
  #+(or)
  (+ 1 1)
  (+ 2 2))
```
or 
```lisp
(defun test (a &optional b)
  #+nil
  (+ 1 1)
  (+ 2 2))
```
or 
```lisp
(defun test (a &optional b)
  #+todo
  (+ 1 1)
  (+ 2 2))
```

__and here is how it looks:__

```lisp
(defun test (a &optional b)
  #+
  (or)
  (+ 1 1)
  (+ 2 2))
```
and
```lisp
(defun test (a &optional b)
  #+nil
  (+ 1 1)
  (+ 2 2))
```
and 
```lisp
(defun test (a &optional b)
  #+todo
  (+ 1 1)
  (+ 2 2))
```

You'll notice the ones on the atoms work as expected.  Some people use `#+nil` or `#+todo` instead of `#+(or)` for this same purpose, so I included all 3 examples.  The value after `#+` can be anything that isn't defined as a feature.  I like `#+(or)` because it is easy to switch to `#+(and)` if I want to enable it temporarily.  When Slime encounters that particular reader macro, it will actually set the color of the s-expression that follows it to the comment color, which is cool, but probably not necessary.  If it kept the `#+` along with the following s-expression or atom immediately after it, that would do it for me.

Sorry if I made this more confusing.  I tend to have an easier time writing things if I have a few examples, so I included more than one.  Let me know if I made it worse.

**Comment by nobody-famous** [comment]
Ok, thanks. That's what I thought it was probably doing. It's treating the open parens as a separator and splitting them. I was going to ask about coloring what's after it as a comment. I can add that, too.

**Comment by nobody-famous** [comment]
Should be fixed in version 0.0.9.

**Comment by smashedtoatoms** [comment]
Thanks for the quick turnaround on this!
